python/pyarrow/_compute.pyx

```diff
@@ -1933,6 +1933,34 @@ class PartitionNthOptions(_PartitionNthOptions):
         self._set_options(pivot, null_placement)
 
 
+cdef class _CumulativeSumOptions(FunctionOptions):
+    def _set_options(self, start, skip_nulls):
+        if not isinstance(start, Scalar):
+            try:
+                start = lib.scalar(start)
+            except Exception:
+                _raise_invalid_function_option(
+                    start, "`start` type for CumulativeSumOptions", TypeError)
+
+        self.wrapped.reset(new CCumulativeOptions(pyarrow_unwrap_scalar(start), skip_nulls))
+
+
+class CumulativeSumOptions(_CumulativeSumOptions):
+    """
+    Options for `cumulative_sum` function.
+
+    Parameters
+    ----------
+    start : Scalar, default 0.0
+        Starting value for sum computation
+    skip_nulls : bool, default False
+        When false, the first encountered null is propagated.
+    """
+
+    def __init__(self, start=0.0, *, skip_nulls=False):
+        self._set_options(start, skip_nulls)
+
```

Comment:
We can go ahead and add the deprecation warning with https://github.com/apache/arrow/blob/f44e28fa03a64ae5b3d9352d21aee2cc84f9af6c/python/pyarrow/util.py#L109

Reply:
Hi, I am not sure how to use `_deprecate_class`, which doesn't look like a decorator.  It seems that we can use like this:

`CumulativeSumOptions = _deprecate_class("CumulativeSumOptions", CumulativeOptions, 14)`

which can't be referenced when using CumulativeSumOptions with pyarrow.compute CumulativeSumOptions.

Reply:
May be we use alias

`Deprecate_CumulativeSumOptions = _deprecate_class("CumulativeSumOptions", CumulativeOptions, 14)` in _compute.py

`Deprecate_CumulativeSumOptions as  CumulativeSumOptions,` in compute.py

Reply:
Hmm, maybe `_deprecate_api` should be used instead then. That does provide a wrapper around the API. If that doesn't work for this use case either, than just a manual warning in the `__init__` method will suffice.

Reply:
I add a manual warning